### DISCUSSION TITLE
cpp/basics/class_and_struct

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#45] [Announcements]
# cpp/basics/class_and_struct

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/basics/class_and_struct.html

**Comment by Qlipphoth** [comment]
终于看到对POD的详细解释了，感谢博主😭

**Comment by 6shuaishuai** [comment]
此处第一行template <class T>中的class和typename等价，表示声明一个模板参数。

**Reply by Mengmengzhizun** [reply]
确实是等价的 但我好奇为什么不用typename而用class  因为少几个单词吗。。。

**Reply by 6shuaishuai** [reply]
参考copilot，在早期的 C++ 标准中，只有 `class` 可以用于模板参数声明。后来，C++ 标准委员会引入了 `typename` 关键字，以便在某些情况下提高代码的可读性和一致性。
